.github/workflows/tests.yml

```diff
@@ -12,6 +12,50 @@ env:
     -Porg.gradle.parallel=false
 
 jobs:
+  # Build Android Tests [Matrix SDK] 
+  build-android-test-matrix-sdk: 
+    name: Matrix SDK - Build Android Tests 
+    runs-on: ubuntu-latest 
+    # No concurrency required, runs every time on a schedule. 
+    steps: 
+      - uses: actions/checkout@v3 
+      - uses: actions/setup-java@v2 
+        with: 
+          distribution: 'adopt' 
+          java-version: 11 
+      - uses: actions/cache@v2 
+        with: 
+          path: | 
+            ~/.gradle/caches 
+            ~/.gradle/wrapper 
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }} 
+          restore-keys: | 
+            ${{ runner.os }}-gradle- 
+      - name: Build Android Tests for matrix-sdk-android 
+        run: ./gradlew clean matrix-sdk-android:assembleAndroidTest $CI_GRADLE_ARG_PROPERTIES --stacktrace 
+ 
+  # Build Android Tests [Matrix APP] 
+  build-android-test-app: 
+    name: App - Build Android Tests 
+    runs-on: ubuntu-latest 
+    # No concurrency required, runs every time on a schedule. 
+    steps: 
+      - uses: actions/checkout@v3 
+      - uses: actions/setup-java@v2 
+        with: 
+          distribution: 'adopt' 
+          java-version: 11 
+      - uses: actions/cache@v2 
+        with: 
+          path: | 
+            ~/.gradle/caches 
+            ~/.gradle/wrapper 
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }} 
+          restore-keys: | 
+            ${{ runner.os }}-gradle- 
+      - name: Build Android Tests for vector 
+        run: ./gradlew clean vector:assembleAndroidTest $CI_GRADLE_ARG_PROPERTIES --stacktrace 
```

Comment:
We can have (or will have) android tests in other module, so I think it should be better to have only one task and call `./gradlew clean assembleAndroidTest` to build tests for every modules (present and future).

I have tested this command on latest develop and have the same expected result:

```
> Task :matrix-sdk-android:compileDebugAndroidTestKotlin FAILED
e: warnings found and -Werror specified
w: /Users/bmarty/workspaces/element-android/matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/E2eeSanityTests.kt: (293, 65): Name shadowed: keysBackupService
w: /Users/bmarty/workspaces/element-android/matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/internal/crypto/E2eeSanityTests.kt: (528, 34): This expression will be resolved to Int in future releases. Please add explicit conversion call
```

Reply:
I think this is fixed in 400a47c